lib/rspec/mocks/space.rb

```diff
@@ -143,10 +143,16 @@ def any_instance_recorders_from_ancestry_of(object)
     private
 
       # We don't want to depend on the stdlib ourselves, but if the user is
-      # using threads then a Mutex will be available to us. If not, we don't
-      # need to synchronize anyway.
+      # using threads then a Mutex will be available to us (which we use via
+      # (RSpec::Support::ReentrantMutex). If not, we don't need to synchronize
+      # anyway.
       def new_mutex
-        defined?(::Mutex) ? ::Mutex.new : FakeMutex
+        if defined?(::Mutex)
```

Comment:
I still think this should be removed because on all rubies besides 1.8.7, `Mutex` is _always_ defined.  On 1.8.7 it's in the stdlib (made available by `require 'thread'`) but on 1.9+ it's a built-in that's always available:

```
➜  ~  ruby --disable=gems -e 'puts Mutex.new'
#<Mutex:0x007fe5940a9048>
```

I (wrongly) thought it would only be available if a user had required `thread` but learned that it's always there as part of rspec/rspec-core#1858.  So now this check is just adding unnecessary overhead and should be removed, IMO.